src/providers/bitcoin/BitcoinSwapProvider.js

```diff
@@ -161,6 +161,15 @@ export default class BitcoinSwapProvider extends Provider {
     return swapTx ? swapTx.hash : null
   }
 
+  async getSecret (claimTxHash) {
+    const claimTxRaw = await this.getMethod('getRawTransactionByHash')(claimTxHash)
+    const claimTx = await this.getMethod('splitTransaction')(claimTxRaw, true)
```

Comment:
you could also just use rpc `decoderawtransaction`

Reply:
Good point, we shouldn't be dependency on `BitcoinLedgerProvider` for something like this.

Reply:
Updated to use `BitcoinRPCProvider`